ms_db/Services/WorkerSubServices.cs

```diff
@@ -0,0 +1,164 @@
+
+namespace ms_db.Services;
+
+public class WorkerSubServices : WorkerBase
+{
+    private readonly myDbContext _dbContext;
+
+    public WorkerSubServices(myDbContext dbContext)
+    {
+        _dbContext = dbContext;
+    }
+
+    private async Task getSchools(string taskId)
+    {
+        var allSchools = await _dbContext.Schools.ToListAsync();
+        List<School> activeSchools = new List<School>();
+        foreach (var school in allSchools)
+        {
+            if (school.ExpiresAt == null)
+            {
+                activeSchools.Add(school);
+            }
+        }
+        var allSchoolsAsJson = JsonSerializer.Serialize(activeSchools);
+        Console.WriteLine($"Pushing the result with id:{taskId}");
+        Console.WriteLine(allSchoolsAsJson);
+        await _db.StringSetAsync(taskId, allSchoolsAsJson);
+        string statusKey = "status" + taskId;
+        await _db.StringSetAsync(statusKey, "Done");
+    }
+
+    private async Task getSchoolById(string taskId,int schoolId)
+    {
+        var currSchool = await _dbContext.Schools.FindAsync(schoolId);
+        
+        var currSchoolAsJson = JsonSerializer.Serialize(currSchool);
+        Console.WriteLine("Found it :"+currSchoolAsJson);
+        await _db.StringSetAsync(taskId, currSchoolAsJson);
+        string statusKey = "status" + taskId;
+        await _db.StringSetAsync(statusKey, "Done");
+    }
+    
+    private async Task getSchoolByName(string taskId,string schoolName)
+    {
+        var currSchool = await _dbContext.Schools.FindAsync(schoolName);
+        var currSchoolAsJson = JsonSerializer.Serialize(currSchool);
+        Console.WriteLine("Found it :"+currSchoolAsJson);
+        await _db.StringSetAsync(taskId, currSchoolAsJson);
+        string statusKey = "status" + taskId;
+        await _db.StringSetAsync(statusKey, "Done");
+    }
+
+    private async Task addNewSchool(string taskId,School newSchool)
+    {
+        /*
+        PropertyInfo[] props = newSchool.GetType().GetProperties();
+        foreach (var type  in props )
+        {
+            Console.WriteLine(type.PropertyType.Name);
+            Console.WriteLine(type.GetValue(newSchool));
+        }*/
+        _dbContext.Schools.Add(newSchool);
+        await _dbContext.SaveChangesAsync();
+        string statusKey = "status" + taskId;
+        await _db.StringSetAsync(statusKey, "Done");
+    }
+
+    private async Task deleteSchool(string taskId,int schoolId)
+    {
+        var deletedSchool = await _dbContext.Schools.FindAsync(schoolId);
+        deletedSchool.ExpiresAt = DateTime.Now.ToString();
+        await _dbContext.SaveChangesAsync();
+        string statusKey = "status" + taskId;
+        await _db.StringSetAsync(statusKey, "Done");
+    }
+
+    private async Task UpdateSchool(string taskId, int schooldId,SchoolUpdateDto updatedSchoolDto)
+    {
+        var updatedSchool = await _dbContext.Schools.FindAsync(schooldId);
+        
+        if (updatedSchoolDto.Name != null )
```

Comment:
Nice that you've validated here also
maybe we should consider creating validation class here as well?